main.odin

```diff
@@ -1186,36 +1192,75 @@ write_docs :: proc(w: io.Writer, docs: string) {
 	start := 0
 	blocks: [dynamic]Block
 
+	example_block: Block // when set the kind should be Example
+	output_block: Block // when set the kind should be Output
+	// rely on zii that the kinds have not been set
+	assert(example_block.kind != .Example)
+	assert(output_block.kind != .Output)
+
+	insert_block :: proc(block: Block, blocks: ^[dynamic]Block, example: ^Block, output: ^Block, name: string) {
+		switch block.kind {
+		case .Paragraph: fallthrough
+		case .Code: append(blocks, block)
+		case .Example:
+			if example.kind == .Example {
+				fmt.eprintf("The documentation for %q has multiple examples which is not allowed\n", name)
+				bad_doc = true
+			}
+			example^ = block
+		case .Output: output^ = block
```

Comment:
`output^ = block` is set twice on this `case`.

Reply:
good catch, silly mistake